package scheduler;

import "DeletePriority.proto";
import "FrequencyOptions.proto";
import "ChannelOptions.proto";
import "EpisodeOptions.proto";

option java_package = "com.comcast.data.protobuf.message.scheduler";
option java_outer_classname = "RecordingOptionsProto";

message RecordingOptions {
  optional EpisodeOptions episodeOptions = 1;
  optional FrequencyOptions frequencyOptions = 2;
  optional ChannelOptions channelOptions = 3;
  optional string channelId = 4;
  required int32 episodesToKeep = 5;
  optional DeletePriority deletePriority = 6;
  required int64 padBeforeSeconds = 7;
  required int64 padAfterSeconds = 8;
}
